### PR TITLE
Fix Test-Json not handling non-object types at root

### LINKED ISSUE #11384: Test-Json doesn't recognize JSON arrays or primitives
# Steps to reproduce

```powershell
'42' | Test-Json | Should -BeTrue
'[ 1, 2 ]' | Test-Json | Should -BeTrue
```

# Expected behavior

The tests should succeed, given that these are valid [JSON strings](https://www.json.org/json-en.html) and that piping them to `ConvertFrom-Json` works just fine:

```powershell
# These tests pass.
'42' | ConvertFrom-Json | Should -Be 42
'[ 1, 2 ]' | ConvertFrom-Json | Should -Be 1, 2
```

# Actual behavior

Both tests fail.

# Environment data

```none
PowerShell Core 7.0.0-rc.1
```

### PR DESCRIPTION
**Pull request by dkaszews**
# PR Summary

`Test-Json` incorrectly parsed JSON as `JObject`, which failed if the root type was not an object, but array, string, number, etc. @mklement0 suggested to change it to `JToken` which matches all types, which I verified to be a one-line fix.

Fixes #11384

## PR Context

`Test-Json` currently fails if the root type is not an object, despite `ConvertFrom-Json` and `ConvertTo-Json` handling them without a problem. This leads to confusing case of `1 | ConvertTo-Json | Test-Json` failing with a non-specific error "Cannot parse JSON".

This bug is most apparent when trying to validate a [JSON schema](https://json-schema.org/) where the root is not a single object, but instead an array of objects. However, as in example above, it also impacts cases without schema, where someone might use `Test-Json` as a shorthand for putting `ConvertFrom-Json` in a try-catch, which is exactly how it should be used when you don't care about content, just whether it is valid.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Test-Json.Tests.ps1 [hunk 1]
```diff
@@ -65,7 +65,7 @@ Describe "Test-Json" -Tags "CI" {
                 errorNode
             }
 "@
-}
+    }
```
Comment (dkaszews) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Test-Json.Tests.ps1:68]: Incorrect indent earlier in the file was confusing my editor, I can remove this change if policy is to minimize diff

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Test-Json.Tests.ps1 [hunk 2]
```diff
@@ -152,4 +152,28 @@ Describe "Test-Json" -Tags "CI" {
         $errorVar[0].FullyQualifiedErrorId | Should -BeExactly "InvalidJsonAgainstSchema,Microsoft.PowerShell.Commands.TestJsonCommand"
         $errorVar[1].FullyQualifiedErrorId | Should -BeExactly "InvalidJsonAgainstSchema,Microsoft.PowerShell.Commands.TestJsonCommand"
     }
+
+    It 'Test-Json recognizes non-object types: <name>' -TestCases @(
+        @{ name = 'number'; value = 1; expected = 'number' }
```
Comment (dkaszews) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Test-Json.Tests.ps1:157]: Test cases from https://github.com/PowerShell/PowerShell/pull/11397 with added schema type verification

**Review by iSazonov — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Review by PaulHigin — APPROVED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by iSazonov** [comment]
@dkaszews Thanks for your contribution!

**Pull request merged**